pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -860,8 +874,8 @@ func (m *ManagerImpl) GetDeviceRunContainerOptions(pod *v1.Pod, container *v1.Co
 		}
 	}
 	if needsReAllocate {
-		klog.V(2).Infof("needs re-allocate device plugin resources for pod %s", podUID)
-		if err := m.allocatePodResources(pod); err != nil {
+		klog.V(2).Infof("needs re-allocate device plugin resources for pod %s, container %s", podUID, container.Name)
+		if err := m.Allocate(pod, container); err != nil {
 			return nil, err
 		}
```

Comment:
It's unclear to me how this will interact with the need for aligned resources across all hint providers. When exactly is this `GetDeviceRunContainerOptions()` function called, and when might it attempt this reallocation of devices?